### PR TITLE
[FLINK-6137][cep] Activate strict checkstyle for flink-cep

### PR DESCRIPTION
**Pull request by dawidwys**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala [hunk 1]
```diff
@@ -271,7 +271,7 @@ class Pattern[T , F <: T](jPattern: JPattern[T, F]) {
     * {{{A1 A2 B}}} appears, this will generate patterns:
     * {{{A1 B}}} and {{{A1 A2 B}}}. See also {{{allowCombinations()}}}.
     *
-    * @return The same pattern with a [[Quantifier.ONE_OR_MORE()]] quantifier applied.
+    * @return The same pattern with a [[Quantifier.oneOrMore()]] quantifier applied.
```
Comment (zentol) [inline flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala]: I hesitated with this change as it is, as i understand it, an API breaking change. @kl0u What's your take on this?

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java [hunk 2]
```diff
@@ -1896,21 +1898,21 @@ public boolean filter(Event value) throws Exception {
...
 	///////////////////////////////         Consecutive           ////////////////////////////////////////
 
 	private static class ConsecutiveData {
-		static final Event startEvent = new Event(40, "c", 1.0);
-		static final Event middleEvent1 = new Event(41, "a", 2.0);
-		static final Event middleEvent2 = new Event(42, "a", 3.0);
-		static final Event middleEvent3 = new Event(43, "a", 4.0);
-		static final Event middleEvent4 = new Event(43, "a", 5.0);
-		static final Event end = new Event(44, "b", 5.0);
+		static final Event START_EVENT = new Event(40, "c", 1.0);
```
Comment (zentol) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java]: you can make these private, then you don't have to rename them.

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPRescalingTest.java [hunk 3]
```diff
@@ -64,7 +68,7 @@ public Integer getKey(Event value) throws Exception {
 
 		Event startEvent1 = new Event(7, "start", 1.0);
 		SubEvent middleEvent1 = new SubEvent(7, "foo", 1.0, 10.0);
-		Event endEvent1=  new Event(7, "end", 1.0);
+		Event endEvent1 =  new Event(7, "end", 1.0);
```
Comment (zentol) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPRescalingTest.java]: while we're at it we can remove the second space after `=`. I remember there were other occurrences of this.

File: flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala (hunk 1 above)
Comment (dawidwys) [inline flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala]: It is not an API change. The Quantifier class is an internal one.

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPRescalingTest.java (hunk 3 above)
Comment (dawidwys) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/operator/CEPRescalingTest.java]: Agree, I tried removing them. I must have missed this one.

File: flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java (hunk 2 above)
Comment (dawidwys) [inline flink-libraries/flink-cep/src/test/java/org/apache/flink/cep/nfa/NFAITCase.java]: Did not know that. Thanks!

File: flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala (hunk 1 above)
Comment (zentol) [inline flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala]: ok then.

Comment (zentol) [inline flink-libraries/flink-cep-scala/src/main/scala/org/apache/flink/cep/scala/pattern/Pattern.scala]: ok then. :)

**Comment by zentol** [comment]
You can rebase this PR now to get rid of the scala import block commit.

**Comment by zentol** [comment]
You'll have to rebase this again to resolve some conflicts, #3978 has fixed a bug in the CEPOperatorTest and hardened it a bit.

The suppressLoadErrors commit has also been merged.

**Comment by zentol** [comment]
I've rebased this locally and will merge this after going through the changes once again.

**Pull request closed without merging**